### ISSUE TITLE
"Operation not permitted"

### ISSUE THREAD

**Issue by frithnanth** [Kaiepi/ra-Trait-Traced#5]
I'm trying to execute the example program in this module's README.md file, but what I get is:
```
1 ROUTINE CALL [1 @ 1582016445.151913]
<== method new (Power)
    self:      (Power)
    $base:     2
    $exponent: 3
    *%_:       {}
==> Power.new(base => 2, exponent => 3)
Operation not permitted
  in method say at /home/user/.raku/sources/59D65395AA9567E09B2BBEA1AC66C77D9681D5EF (Tracer::Default) line 73
  in method trace at /home/user/.raku/sources/94C3987EC49584795FEF2620065B812F59172E7A (Traced) line 60
  in sub new at /home/user/.raku/sources/08D3DB825A55216E688F99DF92EE6871FA1B7596 (Traced::Routine) line 125
  in block <unit> at ./test.raku line 16
```
I'm using Rakudo 2020.01 on Linux Debian sid.
```
$ raku -v
This is Rakudo version 2020.01 built on MoarVM version 2020.01.1
implementing Perl 6.d.
```

**Comment by Kaiepi** [comment]
Sorry for the late response.

This module relies on NativeCall for I/O to standard streams, since I/O with standard streams through `$*OUT`/`$*ERR`/`$*IN` didn't behave well when threads are involved. Could you test 6e489ee4d65a0e3503e59c566f6ff45c4244aaee to see if this is fixed? You'd need to either wait until v2020.03 is released or build Rakudo from master to test it due to recent changes, though.

**Comment by Kaiepi** [comment]
In the meantime, you can trace to a file, which should avoid this error, by placing something like this before any traced calls get made:
```perl6
use Tracer::Default;
$GLOBAL::TRACER = Tracer::Default['.trace'.IO.open: :w];
LEAVE $*TRACER.handle.close;
```

**Comment by Kaiepi** [comment]
The fix for this will be released after v2020.03 of Rakudo gets released.

**Comment by frithnanth** [comment]
OK, thanks.
